catalog/security/catalog-security-xmlattributeplugin/src/main/java/org/codice/ddf/catalog/security/policy/xml/XmlAttributeSecurityPolicyPlugin.java

```diff
@@ -0,0 +1,281 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.security.policy.xml;
+
+import java.io.Serializable;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.StringTokenizer;
+
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamConstants;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+
+import org.codehaus.stax2.XMLInputFactory2;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.operation.Query;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.plugin.PolicyPlugin;
+import ddf.catalog.plugin.PolicyResponse;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.catalog.plugin.impl.PolicyResponseImpl;
+
+/**
+ * Plugin that parses XML metadata for elements that contain attributes with security policy information
+ *
+ */
+public class XmlAttributeSecurityPolicyPlugin implements PolicyPlugin {
+    /**
+     * Logger
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(
+            XmlAttributeSecurityPolicyPlugin.class);
+
+    /**
+     * Input factory
+     */
+    private static volatile XMLInputFactory xmlInputFactory = null;
```

Comment:
If this is `volatile` because you want to ensure happens-before semantics, the static block below should be changed a little bit. You should create a scoped `XMLInputFactory` in the static's scope, set properties on it, and then save it to this static field at the end.